### ISSUE TITLE
Menu shows no user icon because of wrong ampersand encoding in fall-back URL

### ISSUE THREAD

**Issue by viegelinsch** [mautic/mautic#1101]
The user icon in the menu is not displayed if no Gravatar icon can be found, because the fall-back URL is not properly defined. The problem is the encoded ampersand attaching the fall-back URL to the Gravatar URL.

Currently app/bundles/CoreBundle/Templating/Helper/GravatarHelper.php
creates an URL looking like
https://www.gravatar.com/avatar/5c8d0c44de2a05db7875136b0ab4e506?s=250&amp;d=https%3A%2F%2Fwww.example.com%2Fmautic%2Fmedia%2Fimages%2Favatar.png

The exact problem is that the fallback url added in line 72 needs an ampersand as proper URL parameter.

But somewhere the return value runs through an additional urlencode which encodes also the ampersand to "&amp;". Unfortunately I cannot find that spot.

@alanhartless: any idea?

**Comment by alanhartless** [comment]
@viegelinsch Is this still an issue? I'm not able to reproduce this.  In all my Mautic instances local and remote, the ampersand is not encoded.  

Thanks,
Alan

**Comment by escopecz** [comment]
Closing for lack of feedback.

**Comment by rmaddox** [comment]
From last mautic install, the icon is not displaying correctly. It was installed on "localhost", which has its public IP address and the DNS hostname. 

I think Gravatar icon shall not be displayed in Mautic for security. No third party, including Gravatar, shall know where is the database located or Mautic installation installed. The user/owner of Mautic, does not know how many employees are in Gravatar/Wordpress, and which of these employees could be malicious to the server where Mautic has been instaleld.

![Icon not displayed](http://rcdrun.com/images/upload/tmp/2016-01-10-12:22:57-protected.rcdrun.com.png)

**Comment by jotadrilo** [comment]
I faced the same case with a local installation. Gravatar was trying to get a local image as default avatar, which is not accessible by Gravatar.

The current approach only works if the host has a public IP or domain (the file will be accessible and it will be set as the local default avatar), or `devMod` is `true` (the local default avatar will be [mautic.org's default avatar][1]).

Image HTML when Mautic is installed in a local host:
```
<img
src="https://www.gravatar.com/avatar/b58996c504c5638798eb6b511e6f49af?s=250&amp;d=http%3A%2F%2F127.0.0.1%3A8080%2Fmautic%2Fmedia%2Fimages%2Favatar.png" 
pagespeed_url_hash="3594333074" 
onload="pagespeed.CriticalImages.checkImageForCriticality(this);"
>
```

Image HTML when Mautic has `devMode` set to `true`:
```
<img
src="https://www.gravatar.com/avatar/b58996c504c5638798eb6b511e6f49af?s=250&amp;d=https%3A%2F%2Fwww.mautic.org%2Fmedia%2Fimages%2Fdefault_avatar.png" 
pagespeed_url_hash="3594333074" 
onload="pagespeed.CriticalImages.checkImageForCriticality(this);"
>
```

[1]: https://www.mautic.org/media/images/default_avatar.png

**Comment by dongilbert** [comment]
Fixed in #2271
